test/eig.jl

```diff
@@ -8,21 +8,35 @@ using MatrixAlgebraKit: diagview
 @testset "eig_full! for T = $T" for T in (Float32, Float64, ComplexF32, ComplexF64)
     rng = StableRNG(123)
     m = 54
-    for alg in (LAPACK_Simple(), LAPACK_Expert())
+    for alg in
+        (LAPACK_Simple(), LAPACK_Expert(), LAPACK_Simple, LAPACK_Expert, :LAPACK_Simple,
+         :LAPACK_Expert)
```

Comment:
Is the idea that if we only modify this list for the `eig_full` tests, this will test the selection/default functionality and should thus also work for the other factorisations? I am fine with this, and definitely think we should not run all tests 3 times over as this would be a large amount of unnecessary compute. Just checking if this is the plan?

Reply:
Standalone `select_algorithm` and `default_algorithm` tests are definitely a good idea. Regarding the names: I like those names 😄 !